### ISSUE TITLE
getDynamicStyles() includes Observable rules

### ISSUE THREAD

**Issue by bhupinderbola** [cssinjs/jss#930]
__Expected behavior:__
Observable rules should not be included in the getDynamicStyles()

__Describe the bug:__
Observable rules are included in the results of getDynamicStyles()

__Codesandbox link:__ 
https://codesandbox.io/s/xrmo482nr4

__Versions (please complete the following information):__
 - jss: 9.8.7
 - Browser [e.g. chrome, safari]: Chrome
 - OS [e.g. Windows, macOS]: Windows
Feel free to add any additional versions which you may think are relevant to the bug.

**Comment by HenriBeck** [comment]
Can you please explain why this is unexpected? Observables are a form of dynamic styles.

**Comment by kof** [comment]
Or we should fix the problem in react-jss by adding a check for fn values

**Comment by HenriBeck** [comment]
Yeah, we should fix the docs and fix react-jss by adding a check.

Is there a use case for observables inside react-jss currently?

**Comment by HenriBeck** [comment]
But this would break as soon as you render two different instances of the same component.
You would need to create a new HOC for every rendered component.

**Comment by kof** [comment]
Why would it break? We don't reuse dynamic style between elements.

**Comment by HenriBeck** [comment]
Yeah, but the observable would be the same

**Comment by HenriBeck** [comment]
ToDos for this:
- [ ] Update docs to correctly reflect the behavior of `getDynamicStyles`
- [ ] Update react-jss to filter out observables

**Comment by kof** [comment]
True that. I wonder if we should support a fn value that returns an observable.

**Comment by kof** [comment]
Alternatively separate components per animation might be also acceptable.

**Comment by HenriBeck** [comment]
Please no. This would just create more confusion. We would also need a way to pass the observable then back to the user.

We should definitely document the problem with observables and react-jss somewhere.

**Comment by HenriBeck** [comment]
I think for most users function values should be enough.

**Comment by kof** [comment]
Totally, but then we have this Observables and it would be evtl nice to be able to use it with react-jss together. I don't know how to decide on whether it should be supported or not, I think if its not hard and does't causes additional bugs, we could. That thing per component can be solved in user space ... I guess your concern comes from implicitness of this behaviour one wouldn't expect?

**Comment by HenriBeck** [comment]
Yeah, exactly.

**Comment by kof** [comment]
On the other side it would be confusing not having support for observables in react-jss because it claims to support the same syntax, and Observables are part of it.
